src/main/daml/Daml/Finance/Holding/Util.daml

```diff
@@ -5,15 +5,52 @@ module Daml.Finance.Holding.Util where
 
 import DA.Action (foldlA)
 import DA.Assert ((===))
+import DA.Foldable qualified as F (all)
 import DA.Optional (whenSome)
 import DA.Set qualified as S (delete, fromList, insert, intersection, isSubsetOf, notMember, null, singleton, toList)
+import DA.Traversable qualified as T (mapA)
 import Daml.Finance.Interface.Account.Account qualified as Account (Credit(..), Debit(..), I, R, exerciseInterfaceByKey)
 import Daml.Finance.Interface.Account.Util (fetchAccount)
 import Daml.Finance.Interface.Holding.Base qualified as Base (Acquire(..), I, Lock(..), LockType(..), Release(..), getLockers)
+import Daml.Finance.Interface.Holding.Fungible qualified as Fungible (I, Merge(..), Split(..), SplitResult(..))
 import Daml.Finance.Interface.Holding.Transferable qualified as Transferable (I, Transfer(..))
-import Daml.Finance.Interface.Util.Common (fetchInterfaceByKey, qty)
+import Daml.Finance.Interface.Holding.Util (getAmount, getInstrument)
+import Daml.Finance.Interface.Types.Common.Types (InstrumentKey(..))
+import Daml.Finance.Interface.Util.Common (fetchInterfaceByKey, verify, qty)
 import Prelude hiding (null)
 
+-- | Default implementation of split for the Fungible interface.
+splitImpl :
+  ( HasCreate t
+  , HasField "amount" t Decimal
```

Comment:
Since `splitImpl` and `mergeImpl` are only called in one place, why separate them out into their own functions given their relative small size? 

The usage of type class constraints :

- `HasField "amount" t Decimal` (on `splitImpl` and `mergeImpl`);
- `HasField "instrument" t InstrumentKey` (on `mergeImpl`).

This forces any caller to have exactly those fields on their template which makes reuse of these functions arguable.

Reply:
at lease some reuse would be possible, but agree that they are short and can as well live in the `Fungible.T` choice bodies. `getRestAmount` is used in the tests, so left that as a function.